Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/CalculateMonteCarloAbsorptionTest.py

```diff
@@ -0,0 +1,144 @@
+        self._expected_hist = 10
+        self._expected_blocksize = 1905
+
+        self._arguments = {'SampleChemicalFormula': 'H2-O',
+                           'SampleDensityType': 'Mass Density',
+                           'SampleDensity': 1.0,
+                           'EventsPerPoint': 200,
+                           'BeamHeight': 3.5,
+                           'BeamWidth': 4.0,
+                           'Height': 2.0 }
+        self._test_arguments = dict()
+
+    def tearDown(self):
+        DeleteWorkspace(self._red_ws)
+
+        if self._container_ws is not None:
+            DeleteWorkspace(self._container_ws)
+        if self._indirect_elastic_ws is not None:
+            DeleteWorkspace(self._indirect_elastic_ws)
+
+    def _setup_container(self):
+        container_ws = Load('irs26173_graphite002_red.nxs')
+        self._container_ws = container_ws
+
+        container_args = {'ContainerWorkspace':self._container_ws,
+                          'ContainerChemicalFormula':'Al',
+                          'ContainerDensityType':'Mass Density',
+                          'ContainerDensity':1.0 }
+        self._arguments.update(container_args)
+
+    def _setup_flat_plate_container(self):
+        self._setup_container()
+        self._test_arguments['ContainerFrontThickness'] = 1.5
+        self._test_arguments['ContainerBackThickness'] = 1.5
+
+    def _setup_annulus_container(self):
+        self._setup_container()
+        self._test_arguments['ContainerInnerRadius'] = 1.0
+        self._test_arguments['ContainerOuterRadius'] = 2.0
+
+    def _test_corrections_workspace(self, corr_ws):
+        x_unit = corr_ws.getAxis(0).getUnit().unitID()
+        self.assertEquals(x_unit, self._expected_unit)
+
+        y_unit = corr_ws.YUnitLabel()
+        self.assertEquals(y_unit, 'Attenuation factor')
+
+        num_hists = corr_ws.getNumberHistograms()
+        self.assertEquals(num_hists, self._expected_hist)
+
+        blocksize = corr_ws.blocksize()
+        self.assertEquals(blocksize, self._expected_blocksize)
+
+    def _test_corrections_workspaces(self, workspaces):
+        self.assertNotEquals(workspaces, None)
+
+        for workspace in workspaces:
+            self._test_corrections_workspace(workspace)
+
+    def _run_correction_and_test(self, shape):
+        arguments = self._arguments.copy()
+        arguments.update(self._test_arguments)
+        corrected = CalculateMonteCarloAbsorption(SampleWorkspace=self._red_ws,
+                                                  Shape=shape,
+                                                  **arguments)
+        self._test_corrections_workspaces(corrected)
+
+    def _run_correction_with_container_test(self, shape):
+
+        if shape == 'FlatPlate':
+            self._setup_flat_plate_container()
+        else:
+            self._setup_annulus_container()
+
+        self._run_correction_and_test(shape)
+
+    def _run_indirect_elastic_test(self, shape):
```

Comment:
Here you are overriding the member variables, so this tests can be run only after all the others. Moreover you are calling this 3 times now and loading the same file trice, but only the shapes are different. This is not desired. I would suggest to have separate member variables for this case, and set them up also in `setUp`, so the tests are more decoupled.